pkg/model/project.go

```diff
@@ -395,3 +395,57 @@ func ValidateUserGroups(groups []*ProjectUserGroup) error {
 	}
 	return nil
 }
+
+func (p *Project) MergeRBAC() *Project {
```

Comment:
Same as https://github.com/pipe-cd/pipecd/pull/3487/files#r843403939
we can have `ListAllUserGroups() []ProjectUserGroup`.
Wdyt?